scripts/utils/ntc_thermistor_table.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2023 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Zephyr's NTC Thermistor DTS Table generator
+###########################################
+
+This script can be used to generate a "zephyr,thermistor-rt-table" compatible
+Device Tree node for NTC thermistors.  This uses the Beta Parameter Equation
+
+https://devxplained.eu/en/blog/temperature-measurement-with-ntcs#beta-parameter-equation
+
+Look-up the following electrical characteristics in the thermistor's data sheet:
+    Nominal resistance (R25) - The thermistor's resistance measured at 25C
+    Beta value (25/85) - This is the resistance value at 25C and at 85C
+
+Usage::
+
+    python3 ntc_thermistor_table.py
+        -r25 10000
+        -b 3974 > thermistor.dtsi
+
+"""
+
+import argparse
+import numpy as np
+import os
+
+
+def main(
+    r25: float, beta: float, interval: int, temp_init: float, temp_final: float
```

Comment:
Why isn't the interval a `float`?

Reply:
I refactored so `temp_init`, `temp_final`, and `interval` are all `int`.  I didn't see a need to generate a table with that high of a resolution.  By default it will increase by 10 degree Celsius.